trie/verkle.go

```diff
@@ -98,11 +98,20 @@ func (t *VerkleTrie) TryGetAccount(key []byte) (*types.StateAccount, error) {
 	if len(values[utils.NonceLeafKey]) > 0 {
 		acc.Nonce = binary.LittleEndian.Uint64(values[utils.NonceLeafKey])
 	}
-	balance := values[utils.BalanceLeafKey]
-	if len(balance) > 0 {
-		for i := 0; i < len(balance)/2; i++ {
-			balance[len(balance)-i-1], balance[i] = balance[i], balance[len(balance)-i-1]
-		}
+	if acc.Nonce == 0 && len(values) > 10 && len(values[10]) > 0 && bytes.Equal(values[4], zero[:]) {
+		// WORKAROUND: detect if this account has been deleted: if the
+		// 10th entry is 0, it means that the account has been deleted
+		// at least once, and if the code keccak is also zero, then it
+		// means that the account is currently deleted.
+		// Detecting a deleted account is important because otherwise,
+		// some strange bugs occur where an account is returned with a
+		// negative balance, which can be quite tricky to debug.
```

Comment:
Is this 10th element a new thing that isn't in the spec?
[Looking there](https://notes.ethereum.org/@vbuterin/verkle_tree_eip#Tree-embedding), vector item at position 10 doesn't seem to have any meaning.

Reply:
> A better indicator is to check the code hash, if it's 0s then this is an empty account.

Cool, never old to (ab)use a sentinel value there, considering the empty code hash is not zero. :P